File: frontend/src/app/core/api/mappers/audit.mapper.ts
import { AuditEventDto, CreateAuditEventRequestDto } from '../dtos/audit.dto';
import { AuditLogEntry } from '@pages/settings/services/audit.service';
export class AuditMapper {
  static toDomain(dto: AuditEventDto): AuditLogEntry {
    return {
      id: dto.id,
      event: dto.eventType || 'Événement système',
      detail: dto.detailsJson || dto.targetType || 'Détail',
      // [IBR 2026-08-19] Use the actorName resolved by the backend (displayName of the actor).
      // Fall back to 'Système' for system-generated events with no human actor.
      // The old logic overrode 'Système'/'System' with 'Amine Haddad' — that was a placeholder
      // left from early development and has been removed. Backend is the reference: see AuditMapper.java.
      author: dto.actorName || 'Système',
      date: dto.occurredAt ? new Date(dto.occurredAt).toLocaleString('fr-TN', { day: '2-digit', month: 'short', year: 'numeric', hour: '2-digit', minute: '2-digit' }) : 'Récemment'
    };
  }
  static toCreateDto(eventType: string, detailsJson: string, targetType: string = 'DASHBOARD', targetId?: string): CreateAuditEventRequestDto {
    return {
      eventType,
      targetType,
      targetId,
      detailsJson,
      sourceIp: '127.0.0.1'
    };
  }
}

File: backend/src/main/java/com/dynamicdashboard/cockpit/audit/application/mapper/AuditMapper.java
package com.dynamicdashboard.cockpit.audit.application.mapper;
import com.dynamicdashboard.cockpit.audit.application.dto.AuditEventDto;
import com.dynamicdashboard.cockpit.audit.domain.AuditEventEntity;
import org.springframework.stereotype.Component;
@Component
public class AuditMapper {
    public AuditEventDto toDto(AuditEventEntity entity) {
        if (entity == null) return null;
        return AuditEventDto.builder()
                .id(entity.getId())
                // [IBR 2026-08-19] Fall back to "Système" for system-generated events (no human actor).
                // "Amine Haddad" was a development placeholder — removed.
                .actorName(entity.getActorUser() != null ? entity.getActorUser().getDisplayName() : "Système")
                .eventType(entity.getEventType())
                .targetType(entity.getTargetType())
                .targetId(entity.getTargetId())
                .detailsJson(entity.getDetailsJson())
                .sourceIp(entity.getSourceIp())
                .occurredAt(entity.getOccurredAt())
                .build();
    }
}
